src/Microsoft.TemplateEngine.Authoring.TemplateVerifier/VerificationEngine.cs

```diff
@@ -0,0 +1,335 @@
+
+            UsesVerifyAttribute a = new UsesVerifyAttribute();
+            // https://github.com/VerifyTests/Verify/blob/d8cbe38f527d6788ecadd6205c82803bec3cdfa6/src/Verify.Xunit/Verifier.cs#L10
+            //  need to simulate execution from tests
+            var v = DummyMethod;
+            MethodInfo mi = v.Method;
+            a.Before(mi);
+
+            if (_options.VerifyCommandOutput ?? false)
+            {
+                if (Directory.Exists(Path.Combine(commandResult.StartInfo.WorkingDirectory, SpecialFiles.StandardStreamsDir)))
+                {
+                    throw new TemplateVerificationException(
+                        string.Format(
+                            LocalizableStrings.engine_error_stdOutFolderExists,
+                            SpecialFiles.StandardStreamsDir),
+                        TemplateVerificationErrorCode.InternalError);
+                }
+
+                Directory.CreateDirectory(Path.Combine(commandResult.StartInfo.WorkingDirectory, SpecialFiles.StandardStreamsDir));
+
+                await File.WriteAllTextAsync(
+                    Path.Combine(commandResult.StartInfo.WorkingDirectory, SpecialFiles.StandardStreamsDir, SpecialFiles.StdOut + (_options.StandardOutputFileExtension ?? SpecialFiles.DefaultExtension)),
+                    commandResult.StdOut)
+                    .ConfigureAwait(false);
+
+                await File.WriteAllTextAsync(
+                        Path.Combine(commandResult.StartInfo.WorkingDirectory, SpecialFiles.StandardStreamsDir, SpecialFiles.StdErr + (_options.StandardOutputFileExtension ?? SpecialFiles.DefaultExtension)),
+                        commandResult.StdErr)
+                    .ConfigureAwait(false);
+            }
+
+            Task verifyTask = CreateVerificationTask(commandResult.StartInfo.WorkingDirectory, args);
+
+            try
+            {
+                await verifyTask.ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                if (e is TemplateVerificationException)
+                {
+                    throw;
+                }
+                if (e.GetType().Name == "VerifyException")
+                {
+                    throw new TemplateVerificationException(e.Message, TemplateVerificationErrorCode.VerificationFailed);
+                }
+                else
+                {
+                    _logger.LogError(e, LocalizableStrings.engine_error_unexpected);
+                    throw;
+                }
+            }
+        }
+
+        private static class SpecialFiles
+        {
+            public const string StandardStreamsDir = "std-streams";
+            public const string StdOut = "stdout";
+            public const string StdErr = "stderr";
+            public const string DefaultExtension = ".txt";
+            public static readonly string[] FileNames = { StdOut, StdErr };
+        }
+
+        private class LoggerProxy : ITestOutputHelper
```

Comment:
It would be good to get rid of it. I think as we are making own implementation of commands, we can support `ILogger` directly instead (with proxy for tests)